managed/services/dbaas/kubernetes/types.go

```diff
@@ -0,0 +1,336 @@
+	},
+	ClusterTypeGeneric: make(map[string]string),
+}
+
+const (
+	pxcDefaultConfigurationTemplate = `[mysqld]
+wsrep_provider_options="gcache.size=%s"
+wsrep_trx_fragment_unit='bytes'
+wsrep_trx_fragment_size=3670016
+`
+	psmdbDefaultConfigurationTemplate = `
+      operationProfiling:
+        mode: slowOp
+`
+)
+
+func convertComputeResource(res *dbaasv1beta1.ComputeResources) (corev1.ResourceRequirements, error) {
+	req := corev1.ResourceRequirements{}
+	if res == nil {
+		return req, nil
+	}
+	cpu, err := resource.ParseQuantity(fmt.Sprintf("%dm", res.CpuM))
+	if err != nil {
+		return req, err
+	}
+	memory, err := resource.ParseQuantity(strconv.FormatInt(res.MemoryBytes, 10))
+	if err != nil {
+		return req, err
+	}
+	req.Limits = corev1.ResourceList{}
+	req.Limits[corev1.ResourceCPU] = cpu
+	req.Limits[corev1.ResourceMemory] = memory
+	return req, nil
+}
+
+func DatabaseClusterForPXC(cluster *dbaasv1beta1.CreatePXCClusterRequest, clusterType ClusterType) (*dbaasv1.DatabaseCluster, error) {
+	memory := cluster.Params.Pxc.ComputeResources.MemoryBytes
+	gCacheSize := "600M"
+	if cluster.Params.Pxc.Configuration == "" {
+		if memory > memorySmallSize && memory <= memoryMediumSize {
+			gCacheSize = "2.4G"
+		}
+		if memory > memoryMediumSize && memory <= memoryLargeSize {
+			gCacheSize = "9.6G"
+		}
+		if memory > memoryLargeSize {
+			gCacheSize = "9.6G"
+		}
+		cluster.Params.Pxc.Configuration = fmt.Sprintf(pxcDefaultConfigurationTemplate, gCacheSize)
+	}
+	dbCluster := &dbaasv1.DatabaseCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cluster.Name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: dbaasAPI,
+			Kind:       dbaasKind,
+		},
+		Spec: dbaasv1.DatabaseSpec{
+			Database:       databasePXC,
+			DatabaseImage:  cluster.Params.Pxc.Image,
+			DatabaseConfig: cluster.Params.Pxc.Configuration,
+			ClusterSize:    cluster.Params.ClusterSize,
+			DBInstance: dbaasv1.DBInstanceSpec{
+				DiskSize: strconv.FormatInt(cluster.Params.Pxc.DiskSize, 10),
+				CPU:      fmt.Sprintf("%dm", cluster.Params.Pxc.ComputeResources.CpuM),
+				Memory:   strconv.FormatInt(cluster.Params.Pxc.ComputeResources.MemoryBytes, 10),
+			},
+			Monitoring: dbaasv1.MonitoringSpec{
+				PMM: &dbaasv1.PMMSpec{},
+			},
+			LoadBalancer: dbaasv1.LoadBalancerSpec{},
+			Backup:       dbaasv1.BackupSpec{},
+		},
+	}
+	if cluster.Params.Pxc.StorageClass != "" {
+		dbCluster.Spec.DBInstance.StorageClassName = &cluster.Params.Pxc.StorageClass
+	}
+	if cluster.Params.Haproxy != nil {
+		resources, err := convertComputeResource(cluster.Params.Haproxy.ComputeResources)
+		if err != nil {
+			return nil, err
+		}
+		dbCluster.Spec.LoadBalancer.Image = cluster.Params.Haproxy.Image
+		dbCluster.Spec.LoadBalancer.Size = cluster.Params.ClusterSize
+		dbCluster.Spec.LoadBalancer.Resources = resources
+		dbCluster.Spec.LoadBalancer.Type = "haproxy"
+	}
+	if cluster.Params.Proxysql != nil {
```

Comment:
Are we checking somewhere else that we have either `Haproxy` or `Proxysql`, just one of them?
I don't remember if we are doing that in another place.

Reply:
No. Great catch. I'll update it soon

Reply:
Fixed it. Thanks